### DISCUSSION TITLE
WIX Integration?

### DISCUSSION THREAD

**Discussion by juansgalt** [btcpayserver/btcpayserver#967] [Plugins & integrations]
Hello, 

Is it possible and anywhere in the horizon to add integration for WIX? 

Compared to alternatives, Wix is super easy to design websites with, and it is integrated with major players through the internet. 

Currently, I'm forced to go through Ecwid.com and then Coinpayments.com which is a centralized solution, I'd love to integrate BTCPAY but so far finding no easy way to do it that flows with the shopping experience.

In my case it is for a phone accessory. It would be great if BTCPAY integrated with the shopping cart itself, similar to how Paypal does. But there are probably multiple ways that it could work.

The key feature I need is for btcpay to recognize what is in the wix shopping cart.

I'm not an expert at this which is in part why this feature might be valuable to small businesses and so on. 

Curious to hear what the challenges here are and what to expect.

**Comment by woutersamaey** [comment]
For this kind of thing I would suggest that we build a pure-JavaScript solution that can be used on any (simple) website.

The JavaScript needs to be told what the input for the invoice is so it can read the data from the page and then it must be told where to render a "Pay with BTCPay" button. Nothing more.

This would use the less secure "allow anyone to create invoices" option, but okay, it's for simple use cases. You can't make this secure without server side code.

The Javacript could look like this to implement:

```
BTCPayButton.init({
   customer_name: '.customer-name',
   customer_email: '.customer-email',
   total_price: '.totals .grand_total',
   currency_code: '.totals .currency-code'
   ...
   render_pay_button: '.pay-button-placeholder',
   redirect_to_url_on_success: 'thank-you.html'
});
```

Can be adapted to almost any simple website, though some sites may need to add some HTML if the page does not include the customer email etc...

Of course, there wouldn't be any IPN notifications.

**Comment by pavlenex** [comment]
https://docs.btcpayserver.org/Wix/ wix integration is now available
